host/src/server/api/v2/proof/mod.rs

```diff
@@ -39,6 +44,49 @@ async fn proof_handler(State(actor): State<Actor>, Json(req): Json<Value>) -> Ho
     let mut config = actor.default_request_config().clone();
     config.merge(&req)?;
 
+    // For zk_any request, draw zk proof type based on the block hash.
+    //
+    // A zk_any request looks like: { "proof_type": "zk_any", "zk_any": { "aggregation": <bool> } }
+    let is_zk_any = config.proof_type == Some("zk_any".to_string());
+    if is_zk_any {
+        let network = config
+            .network
+            .as_ref()
+            .ok_or(RaikoError::InvalidRequestConfig(
+                "Missing network".to_string(),
+            ))?;
+        let block_number = config.block_number.ok_or(RaikoError::InvalidRequestConfig(
+            "Missing block number".to_string(),
+        ))?;
+        let (_, blockhash) = get_task_data(&network, block_number, actor.chain_specs()).await?;
+        match actor.draw(&blockhash) {
+            Some(proof_type) => config.proof_type = Some(proof_type.to_string()),
+            None => {
+                return Ok(Status::Ok {
+                    proof_type: ProofType::Native,
+                    data: ProofResponse::Status {
+                        status: TaskStatus::ZKAnyNotDrawn,
```

Comment:
do we need to store this status into db?? so it will tell the caller that this one has already missed the draw. So far our fixed algorithm selection is ok, probably someday we use a random based selection, then repeated callings would eventually hit. however, I don't think we will use random based draw in near future.